devito/arch/compiler.py

```diff
@@ -771,6 +771,18 @@ def __init_finalize__(self, **kwargs):
             self.__init_intel_mpi__()
             self.__init_intel_mpi_flags__()
 
+        if configuration['profiling'] == 'advisor':
```

Comment:
see comment below, but basically since "profiling" requires third-party tools (as opposed to MPI which is an actual API) I have a preference for this bit of logic to be kept in profiling